eventlet/greenthread.py

```diff
@@ -206,7 +208,7 @@ def unlink(self, func, *curried_args, **curried_kwargs):
 
         Remove successfully return True, otherwise False
         """
-        if not getattr(self, '_exit_funcs', None):
+        if not self._exit_funcs:
```

Comment:
This early exit could vanish too. Once `_exit_funcs` is always a `deque`, removing this test would optimize valid `unlink()` calls rather than optimizing invalid calls.